### PR TITLE
Multiroot improvements

### PR DESCRIPTION
**Pull request by escaton**
- [x] ~add start-all, stop-all, restart{-all} commands~
- [x] handle `disabledWorkspaceFolders` change on the fly
- [x] misc: use Map to store jest instances

#trivial

### CODE REVIEW

**Comment by escaton** [comment]
@connectdotz please take a look.
Unfortunately i noticed too late that there was #441
@vdh you've done great research and i would love to drop my commit in favor of your solution, but it would require some refactor from you. I'd suggest you to rebase to `beta` branch and check out the changes.

File: src/extensionManager.ts [hunk 1]
```diff
@@ -53,42 +54,51 @@ export class ExtensionManager {
...
-    return (
-      // start if enabledWorkspaceFolders contains current workspace
-      enabledWorkspaceFolders.includes(workspaceFolderName) ||
-      // dont start if disabledWorkspaceFolders doesn't contain current workspace
-      !disabledWorkspaceFolders.includes(workspaceFolderName)
-    )
+    const { commonPluginSettings: { disabledWorkspaceFolders } } = this
+    if (this.extByWorkspace.has(workspaceFolderName)) {
+      return false
+    } else if (disabledWorkspaceFolders.includes(workspaceFolderName)) {
```
Comment (connectdotz) [inline src/extensionManager.ts]: no need for `else`,  you can also consider consolidating these 2 conditions into a single if statement, might be more readable...

**Comment by connectdotz** [comment]
probably update changelog since many users have been asking for these new commands

**Comment by escaton** [comment]
Yes, but i’m considering remove that commit and let @vdh to update his PR. So i guess i’ll add `#trivial`

File: src/extensionManager.ts (hunk 1 above)
Comment (escaton) [reply src/extensionManager.ts]: I did this with the same motivation :)

Comment (connectdotz) [reply src/extensionManager.ts]: what is the reason for `else`?

Comment (escaton) [reply src/extensionManager.ts]: Ah, missed that. No reason for `else`, i'll split it in two separate conditions.

**Comment by vdh** [comment]
@escaton I've rebased the PR, although since this is my first contribution it'd be great to get someone double check the logic of the Promise-based changes I've made.

**Comment by escaton** [comment]
Dropped my commits related to start/stop/restart commands

**Comment by Bnaya** [comment]
Is there any overview of the features here? 
Thanks ! :)

**Comment by connectdotz** [comment]
> Is there any overview of the features here?

there is a brief introduction in the [README.md](https://github.com/jest-community/vscode-jest/blob/beta/README.md#use-extension-in-multiroot-environment), if it's not intuitive then maybe we should consider adding more...

**Comment by escaton** [comment]
@connectdotz would you like to merge this PR to `beta`?

**Comment by connectdotz** [comment]
yes, was waiting for the restart-command PR, now it is in, is there anything you want to change in this PR before I merge it?

**Comment by escaton** [comment]
I'm done :)

**Pull request merged**